packages/discovery-react-components/src/utils/document/transformEnrichment.ts

```diff
@@ -38,46 +52,35 @@ function setMetadata(enrichedHtml: Contract): void {
       });
     }
   });
+
   enrichedHtml.metadata = updatedMetadata;
-}
 
-function getOntology(enrichmentName: string): Ontology | undefined {
-  let ontology;
-  if (enrichmentName === ENRICHMENTS.INVOICE) {
-    ontology = invoiceOntology;
-  } else if (enrichmentName === ENRICHMENTS.PURCHASE_ORDER) {
-    ontology = purchaseOrderOntology;
-  }
-  return ontology;
+  return enrichedHtml;
 }
 
-function setAttributesAndRelations(doc: any, enrichmentName: string): void {
-  const ontology = getOntology(enrichmentName);
-  if (ontology) {
-    doc.attributes = createAttributeObjects(ontology.attributes, doc);
-    doc.relations = createRelationObjects(ontology, doc);
-  }
+function setAttributesAndRelations(
+  enrichment: Invoice | PurchaseOrder,
+  enrichmentName: string
+): Invoice | PurchaseOrder {
+  const ontology = ontologyMapping[enrichmentName];
+  enrichment.attributes = setAttributes(ontology, enrichment);
```

Comment:
Ideally, we make the input objects immutable, and return a new object with any data adjustments that we need.

Reply:
We already making a cloned copy of EnrichedHtml in processDoc so not doing it again here in transformEnrichment file.

Reply:
It's more about the principle of immutability that I'm concerned with. We shouldn't be making changes to input arguments or using those variables for anything that isn't read-only within the function.

Reply:
I was doing that initially, but based on the comment, changed that back https://github.com/watson-developer-cloud/discovery-components/pull/69#discussion_r370415062

Reply:
Ah, didn't see that. I guess I'm overruled. I really would prefer immutability, but I'll bow out to @jhpedemonte.

Reply:
Yeah, immutability is a larger discussion to be had.